### PR TITLE
Add GroupState for group management

### LINKED ISSUE #23477: Group state is managed with Zeebe
<h2>Description</h2>

We need to introduce an `GroupState` to Zeebe in which we can store Groups.

1. Create a `DbGroupState` which implements the `MutableGroupState` interface which implements the `GroupState` interface.
    - Look at other state classes for examples. 
2. Create ColumnFamily `GROUPS`
    - Add it to the `ZbColumnFamilies` enum
    -  Define it in the `DbGroupState` class:
        - Define a `DbLong groupKey` 
        - Define a `PersistedGroup persistedGroup`
            - `PersistedGroup` contains two properties:
                - groupKey
                - name
        - Define the ColumnFamily:
            - Key: `groupKey`
            - Value: `persistedGroup`
3. Create ColumnFamily `ENTITY_BY_GROUP`
    - Add it to the `ZbColumnFamilies` enum
    -  Define it in the `DbGroupState` class:
        - Define a `DbForeignKey fkGroupKey`
        - Define a `DbLong entityKey` 
        - Define a `EntityType entityType`
            - `EntityTypeValue` contains properties:
                - `entityType` - `EnumProperty` (take inspiration from `JobStateValue`)
        - Define the ColumnFamily:
            - Key: composite key of `fkGroupKey` and `entityKey`
            - Value: `entityType`
4. Create ColumnFamily `GROUP_BY_NAME`
    - Add it to the `ZbColumnFamilies` enum
    - Define it in the `DbGroupState` class:
        - Define a `DbString groupName`
        - Define the ColumnFamily:
            - Key: `groupName`
            - Value: `fkGroupKey`

### PR DESCRIPTION
**Pull request by koevskinikola**
## Description

Define basic Zeebe state interfaces and classes for Group management and link them to the `ProcessingState` classes. Basic `DbGroupState` properties and column families are also defined.

## Checklist

- [ ] for CI changes:
  - [ ] structural/foundational changes signed off by [CI DRI](https://github.com/cmur2)
  - [ ] [ci.yml](https://github.com/camunda/camunda/blob/main/.github/workflows/ci.yml) modifications comply with ["Unified CI" requirements](https://github.com/camunda/camunda/wiki/CI-&-Automation#workflow-inclusion-criteria)

## Related issues

closes #23477

### CODE REVIEW

File: zeebe/engine/src/main/java/io/camunda/zeebe/engine/state/group/PersistedGroup.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+import io.camunda.zeebe.db.DbValue;
+import io.camunda.zeebe.msgpack.UnpackedObject;
+import io.camunda.zeebe.msgpack.property.LongProperty;
+import io.camunda.zeebe.msgpack.property.StringProperty;
+import io.camunda.zeebe.util.buffer.BufferUtil;
+
+public class PersistedGroup extends UnpackedObject implements DbValue {
+
+  private final LongProperty groupKeyProp = new LongProperty("groupKey", -1L);
+  private final StringProperty nameProp = new StringProperty("name", "");
```
Comment (nicpuppa) [inline zeebe/engine/src/main/java/io/camunda/zeebe/engine/state/group/PersistedGroup.java:19]: 💭 I think this 2 properties shouldn't be optional, but we can change them later on 😄

**Review by nicpuppa — APPROVED** [review]
🚀

File: zeebe/engine/src/main/java/io/camunda/zeebe/engine/state/group/PersistedGroup.java (hunk 1 above)
Comment (koevskinikola) [reply zeebe/engine/src/main/java/io/camunda/zeebe/engine/state/group/PersistedGroup.java:19]: That's a good point. I was going through our persisted objects and we are pretty inconsistent in how we set these values, and even if we store the whole `*Record` object directly, or if we only mirror certain properties.

🤔 Do we have any guides on how to determine what approach should be taken?

Comment (koevskinikola) [reply zeebe/engine/src/main/java/io/camunda/zeebe/engine/state/group/PersistedGroup.java:19]: @nicpuppa I can update them now. Should I also update the `PersistedRole? The properties are also optional there.

Comment (nicpuppa) [reply zeebe/engine/src/main/java/io/camunda/zeebe/engine/state/group/PersistedGroup.java:19]: I have a task on my todo list to review all this properties, I think can be done later. But if you are up to it go ahead 😄 Also `PersistedMapping` and `PersistedTenant` should be checked

Comment (nicpuppa) [reply zeebe/engine/src/main/java/io/camunda/zeebe/engine/state/group/PersistedGroup.java:19]: > 🤔 Do we have any guides on how to determine what approach should be taken?

I'm not sure, but we should have in the persisted object only what is needed, sometimes doesn't make sense to have all the property of the record, if the persisted record is just a wrapper of the record, what's the point to have it 😄

Comment (koevskinikola) [reply zeebe/engine/src/main/java/io/camunda/zeebe/engine/state/group/PersistedGroup.java]: I updated the `PersistedGroup`, `PersistedRole`, and `PersistedTenant` classes and removed the default values of the requited properties.

Note that for `PersistedTenant` the `name` property isn't required (at least it isn't used as a column key).

**Review by nicpuppa — APPROVED** [review]

**Pull request merged**